### PR TITLE
fix(dot/core): Fix handle transaction message test.

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Fix handle transaction message test

## Tests

```
TestService_HandleTransactionMessage
```

## Issues

-

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]

File: dot/core/messages_test.go [hunk 1]
```diff
@@ -91,16 +80,23 @@ func TestService_HandleTransactionMessage(t *testing.T) {
...
+	parentHash := common.MustHexToHash("0x35a28a7dbaf0ba07d1485b0f3da7757e3880509edc8c31d0850cb6dd6219361d")
+	header, err := types.NewHeader(parentHash, common.Hash{}, common.Hash{}, big.NewInt(1), types.NewEmptyDigest())
+	require.NoError(t, err)
+
+	// initialise block header
+	err = s.rt.InitializeBlock(header)
+	require.NoError(t, err)
+
+	ext := types.Extrinsic(common.MustHexToBytes("0x410284ffd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d015a3e258da3ea20581b68fe1264a35d1f62d6a0debb1a44e836375eb9921ba33e3d0f265f2da33c9ca4e10490b03918300be902fcb229f806c9cf99af4cc10f8c0000000600ff8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a480b00c465f14670"))
```
Comment (noot) [inline dot/core/messages_test.go:99]: where did this come from?

File: dot/core/messages_test.go [hunk 2]
```diff
@@ -91,16 +80,23 @@ func TestService_HandleTransactionMessage(t *testing.T) {
...
 		TransactionState: state.NewTransactionState(),
 		IsBlockProducer:  true,
+		BlockProducer:    &mockBlockProducer{},
 	}
 
 	s := NewTestService(t, cfg)
 
-	// https://github.com/paritytech/substrate/blob/5420de3face1349a97eb954ae71c5b0b940c31de/core/transaction-pool/src/tests.rs#L95
-	ext := []byte{1, 212, 53, 147, 199, 21, 253, 211, 28, 97, 20, 26, 189, 4, 169, 159, 214, 130, 44, 133, 88, 133, 76, 205, 227, 154, 86, 132, 231, 165, 109, 162, 125, 142, 175, 4, 21, 22, 135, 115, 99, 38, 201, 254, 161, 126, 37, 252, 82, 135, 97, 54, 147, 201, 18, 144, 156, 178, 38, 170, 71, 148, 242, 106, 72, 69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 216, 5, 113, 87, 87, 40, 221, 120, 247, 252, 137, 201, 74, 231, 222, 101, 85, 108, 102, 39, 31, 190, 210, 14, 215, 124, 19, 160, 180, 203, 54, 110, 167, 163, 149, 45, 12, 108, 80, 221, 65, 238, 57, 237, 199, 16, 10, 33, 185, 8, 244, 184, 243, 139, 5, 87, 252, 245, 24, 225, 37, 154, 163, 142}
+	parentHash := common.MustHexToHash("0x35a28a7dbaf0ba07d1485b0f3da7757e3880509edc8c31d0850cb6dd6219361d")
```
Comment (noot) [inline dot/core/messages_test.go:91]: same here, where did this come from? it would be nicer to generate these in the test instead of hard coding

Comment (arijitAD) [reply dot/core/messages_test.go:91]: I generated an extrinsic to transfer a token from Alice to Bob. I will add the extrinsic generation code.

File: dot/core/messages_test.go [hunk 3]
```diff
@@ -75,40 +74,81 @@ func TestService_ProcessBlockAnnounceMessage(t *testing.T) {
 	require.Equal(t, expected, net.Message)
 }
 
-func TestService_HandleTransactionMessage(t *testing.T) {
-	// this currently fails due to not being able to call validate_transaction
+func createExtrinsics(t *testing.T, rt runtime.Instance, genHash common.Hash, nonce uint64) types.Extrinsic {
+	t.Helper()
+	bob, err := ctypes.NewAddressFromHexAccountID("0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
```
Comment (noot) [inline dot/core/messages_test.go:79]: this is fine, but you can also get the key with `keystore.NewSr25519Keyring().Bob().Public().Hex()`

**Review by noot — APPROVED** [review]

File: dot/core/messages_test.go (hunk 3 above)
Comment (Tikaryan) [reply dot/core/messages_test.go:79]: Done

Comment (arijitAD) [reply dot/core/messages_test.go]: Done

**Pull request merged**